client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -414,6 +415,18 @@ public static ContentType createContentType(final XContentType xContentType) {
         return ContentType.create(xContentType.mediaTypeWithoutParameters(), (Charset) null);
     }
 
+    static Request indicesExist(IndicesExistsRequest request) {
+        String endpoint = endpoint(request.indices(), Strings.EMPTY_ARRAY, "");
+        Params params = Params.builder();
+        params.putParam("local", Boolean.FALSE.toString());
+        params.putParam("ignore_unavailable", String.valueOf(request.indicesOptions().ignoreUnavailable()));
+        params.putParam("allow_no_indices", String.valueOf(request.indicesOptions().allowNoIndices()));
+        params.putParam("expand_wildcards", "open");
+        params.putParam("flat_settings", Boolean.FALSE.toString());
+        params.putParam("include_defaults", Boolean.FALSE.toString());
```

Comment:
would you mind adding withFlatSettings and withIncludeDefaults methods ? they will be used from other APIs in the future.

Reply:
Definitely don't mind. Is something like this good:
```
Params withFlatSettings(boolean flatSettings) {
            if (flatSettings) {
                return putParam("flat_settings", Boolean.TRUE.toString());
            }
            return this;
        }
```
In other words, is the default value for all request types the same?

Reply:
I think so, this should be good enough.